burn-wgpu/src/kernel/binary_elemwise.rs

```diff
@@ -72,65 +74,109 @@ pub fn binary_elemwise<K: StaticKernel, E: WgpuElement, const D: usize>(
         });
 
     let shape_out = Shape::new(shape_out);
+    let num_elems = shape_out.num_elements();
 
     let buffer = lhs
         .context
-        .create_buffer(shape_out.num_elements() * core::mem::size_of::<E>());
+        .create_buffer(num_elems * core::mem::size_of::<E>());
     let output = WgpuTensor::new(lhs.context.clone(), shape_out, buffer);
 
     let kernel = lhs
         .context
-        .compile_static::<KernelSettings<K, E, i32, WORKGROUP, 1, 1>>();
+        .compile_static::<KernelSettings<K, E, i32, WORKGROUP, WORKGROUP, 1>>();
     let info = build_info(&[&lhs, &rhs, &output]);
     let info_buffers = lhs
         .context
         .create_buffer_with_data(bytemuck::cast_slice(&info));
 
     lhs.context.execute(
-        WorkGroup::new(
-            f32::ceil(output.shape.num_elements() as f32 / WORKGROUP as f32) as u32,
-            1,
-            1,
-        ),
+        elemwise_workgroup(num_elems, WORKGROUP),
         kernel,
         &[&lhs.buffer, &rhs.buffer, &output.buffer, &info_buffers],
     );
 
     output
 }
-pub fn binary_elemwise_inplace<K: StaticKernel, E: WgpuElement, const D: usize>(
+
+/// Execute a binary inplace kernel using the default settings.
+pub fn binary_elemwise_inplace_default<K: StaticKernel, E: WgpuElement, const D: usize>(
     lhs: WgpuTensor<E, D>,
     rhs: WgpuTensor<E, D>,
 ) -> WgpuTensor<E, D> {
-    lhs.assert_is_on_save_device(&rhs);
+    binary_elemwise_inplace::<K, E, D, 32>(lhs, rhs)
+}
 
-    let mut shape_out = [0; D];
-    lhs.shape
-        .dims
-        .iter()
-        .zip(rhs.shape.dims.iter())
-        .enumerate()
-        .for_each(|(index, (dim_lhs, dim_rhs))| {
-            shape_out[index] = usize::max(*dim_lhs, *dim_rhs);
-        });
+/// Execute a binary inplace kernel using the provided WORKGROUP.
+pub fn binary_elemwise_inplace<
+    K: StaticKernel,
+    E: WgpuElement,
+    const D: usize,
+    const WORKGROUP: usize,
+>(
+    lhs: WgpuTensor<E, D>,
+    rhs: WgpuTensor<E, D>,
+) -> WgpuTensor<E, D> {
+    lhs.assert_is_on_save_device(&rhs);
```

Comment:
Should this be `assert_is_on_same_device` instead of `assert_is_on_save_device`? Maybe we should rename the function name.